<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="front_cam_topic_uncompressed" default="/sensors/camera/front/image_raw"/>
    <arg name="front_cam_topic_compressed" default="/sensors/camera/front/image_raw/compressed"/>
    <arg name="down_cam_topic_uncompressed" default="/sensors/camera/down/image_raw"/>
    <arg name="down_cam_topic_compressed" default="/sensors/camera/down/image_raw/compressed"/>
    <arg name="front_enhanced_topic" default="/vision/front_cam/image_enhanced"/>
    <arg name="down_enhanced_topic" default="/vision/down_cam/image_enhanced"/>
    <arg name="sim" default="false"/>

    <let name="front_cam_topic" value="$(var front_cam_topic_compressed)" if="$(var sim)"/>
    <let name="front_cam_topic" value="$(var front_cam_topic_uncompressed)" unless="$(var sim)"/>
    <let name="down_cam_topic" value="$(var down_cam_topic_compressed)" if="$(var sim)"/>
    <let name="down_cam_topic" value="$(var down_cam_topic_uncompressed)" unless="$(var sim)"/>
    <node pkg="vision" exec="front_image_enhancement" name="front_image_enhancement" output="screen">
        <remap from="/zed/zed_node/left/image_rect_color" to="$(var front_cam_topic)"/>
        <param name="input_topic" value="$(var front_cam_topic)"/>
        <param name="output_topic" value="$(var front_enhanced_topic)"/>
        <param name="sim" value="$(var sim)"/>
    </node>

    <node pkg="vision" exec="down_image_enhancement" name="down_image_enhancement" output="screen">
        <param name="input_topic" value="$(var down_cam_topic)"/>
        <param name="output_topic" value="$(var down_enhanced_topic)"/>
        <param name="sim" value="$(var sim)"/>
    </node>
</launch>